Code/Framework/AzToolsFramework/AzToolsFramework/ComponentModes/BoxViewportEdit.h

```diff
@@ -18,21 +18,33 @@ namespace AzToolsFramework
 
     /// Wraps 6 linear manipulators, providing a viewport experience for 
     /// modifying the extents of a box
-    class BoxViewportEdit : public BaseViewportEdit
+    class BoxViewportEdit : public BaseShapeViewportEdit
     {
     public:
         BoxViewportEdit(bool allowAsymmetricalEditing = false);
 
-        // BaseViewportEdit overrides ...
-        void Setup(const AZ::EntityComponentIdPair& entityComponentIdPair) override;
+        // BaseShapeViewportEdit overrides ...
+        void Setup() override;
         void Teardown() override;
         void UpdateManipulators() override;
         void ResetValues() override;
+        void AddEntityComponentIdPair(const AZ::EntityComponentIdPair& entityComponentIdPair) override;
+
+        void InstallGetBoxDimensions(const AZStd::function<AZ::Vector3()>& getBoxDimensions);
+        void InstallGetLocalTransform(const AZStd::function<AZ::Transform()>& getLocalTransform);
+        void InstallSetBoxDimensions(const AZStd::function<void(const AZ::Vector3)>& setBoxDimensions);
 
     private:
-        AZ::EntityComponentIdPair m_entityComponentIdPair;
+        AZ::Vector3 GetBoxDimensions() const;
+        AZ::Transform GetLocalTransform() const;
+        void SetBoxDimensions(const AZ::Vector3& boxDimensions);
+
         using BoxManipulators = AZStd::array<AZStd::shared_ptr<LinearManipulator>, 6>;
         BoxManipulators m_linearManipulators; ///< Manipulators for editing box size.
         bool m_allowAsymmetricalEditing = false; ///< Whether moving individual faces independently is allowed.
+
+        AZStd::function<AZ::Vector3()> m_getBoxDimensions;
+        AZStd::function<AZ::Transform()> m_getLocalTransform;
+        AZStd::function<void(const AZ::Vector3)> m_setBoxDimensions;
```

Comment:
nit: missing `&` -> `const AZ::Vector3&` (also consider `using` idea mentioned earlier)

Reply:
+1

Reply:
Added the &, but still not keen on the using idea

Reply:
Sure no worries 👍